torch/autograd/gradcheck.py

```diff
@@ -613,10 +845,11 @@ def gradcheck(
     rtol: float = 1e-3,
     raise_exception: bool = True,
     check_sparse_nnz: bool = False,
-    nondet_tol: float = 0.0,
+    nondet_tol: float = 1e-12,  # TODO WHY
```

Comment:
Failures without the default of 1e-12 include: 
 - cummin/cummax the tests for this op are in the progress of being ported to OpInfos https://github.com/pytorch/pytorch/pull/54762. We can specify cummin/cummax to use slow gradcheck or specifically set its nondet tol to 1e-12.